test_extract_unique_sensors_01.py

```diff
@@ -0,0 +1,65 @@
+import unittest
+import logging
+from mock import MagicMock
+from mock import patch
+from urllib.request import urlopen
+from extract_unique_sensors_01 import read_sensors
+from extract_unique_sensors_01 import extract_all_sensors
+from extract_unique_sensors_01 import splitting_extracted_sensors
+
+
+class TestExtractSensors(unittest.TestCase):
+    """Test class for function test_read_sensors."""
+
+    def test_read_sensors(self):
+        with patch('urllib.request.urlopen') as mock_urlopen:
+            url = "http://lebogang.com"
+            mock_urlopen.return_value.status_code = 200
+            self.assertTrue(url)
+
+        with patch('urllib.request.urlopen') as mock_urlopen:
+            non_existing_url = "http://google.com/fakeurl"
+            mock_urlopen.return_value.status_code = 404
+            self.assertTrue(non_existing_url)
```

Comment:
@amakhaba Please assist here with the mocks.

You could do something like:
```python
        file_name = MagicMock()
        url = MagicMock()
        DUT = read_sensors(file_name, url)
```

Where DUT is the device under test, and you could interact with it similar to what you did in line #52 and line 64

I cannot think of a better solution for now, but if you need assistance let me know.